rasa/nlu/training_data/training_data.py

```diff
@@ -75,12 +75,18 @@ def merge(self, *others: "TrainingData") -> "TrainingData":
             nlg_stories,
         )
 
-    def filter_by_intent(self, intent: Text):
-        """Filter training examples """
+    def filter_training_examples(
+        self, condition: Callable[[Message], bool]
+    ) -> "TrainingData":
+        """Filter training examples.
+
+        Args:
+            - condition: A function that will be applied to filter training examples.
+        """
 
         training_examples = []
```

Comment:
Can we just use a filter instead of this loop?

```suggestion
        training_examples = list(filter(condition, training_examples))
```